src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -64,6 +64,8 @@ public static void Resize<T>([NotNull] ref T[]? array, int newSize)
                 Copy(larray, 0, newArray, 0, larray.Length > newSize ? newSize : larray.Length);
                 array = newArray;
             }
+
+            NullabilityHelpers.SuppressNonNullAssignmentWarning(ref array);
```

Comment:
```suggestion
            Debug.Assert(array != null);
```

Reply:
This won't solve the problem. The compiler expects to see a write to the _array_ parameter on every code path, or it expects the _array_ parameter to be passed to a method whose ref parameter is marked `[NonNull]`. I couldn't figure out any way to solve this other than by using a helper method as introduced here (which gets compiled away entirely due to the `[Conditional]` attribute) or by using a `#pragma warning disable CS8777` at the call site.

Reply:
> This won't solve the problem.

Are you sure?  I just tried it with last night's SDK and it appears to.  Generally `Debug.Assert` can be used to tell the compiler that something is non-null because the Assert argument is attributed as `DoesNotReturnIf(false)`

Reply:
> Derp. I typo good in source.

:smile: